### PR TITLE
Fix invalid references for /server/solr

### PR DESCRIPTION
**Pull request by jebnix**
…a documentation

https://issues.apache.org/jira/browse/SOLR-XXXXX

# Description

NOTE - I'm not sure if this fix is correct. This is tightly related to https://github.com/apache/solr/pull/1109.
This part references that Solr's data is kept in the path of `/server/solr`, which isn't mentioned in any other place I've found. I think that this is a mistake and the author meant `/opt/solr/server/solr`. If I'm correct, this path is no longer used in favor of `/var/solr/data`.

# Solution

Understanding what is the correct path that the author meant, and fixing it if it is needed.

# Tests

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [x] I have added tests for my changes.
- [x] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

File: solr/solr-ref-guide/modules/deployment-guide/pages/docker-faq.adoc [hunk 1]
```diff
@@ -48,8 +48,8 @@ This is useful if you want to inspect or modify the data in the Docker host
 when the container is not running, and later easily run new containers against that data.
 This is indeed possible, but there are a few gotchas.
 
-Solr stores its core data in the `server/solr` directory, in sub-directories for each core.
-The `server/solr` directory also contains configuration files that are part of the Solr distribution.
+Solr stores its core data in the `var/solr/data` directory, in sub-directories for each core.
```
Comment (janhoy) [inline solr/solr-ref-guide/modules/deployment-guide/pages/docker-faq.adoc:51]:
```suggestion
Solr stores its core data in the `/var/solr/data` directory, in sub-directories for each core.
```

File: solr/solr-ref-guide/modules/deployment-guide/pages/docker-faq.adoc [hunk 2]
```diff
@@ -48,8 +48,8 @@ This is useful if you want to inspect or modify the data in the Docker host
...
+The `var/solr/data` directory also contains configuration files that are part of the Solr distribution.
```
Comment (janhoy) [inline solr/solr-ref-guide/modules/deployment-guide/pages/docker-faq.adoc:52]: I'm not sure this is as true for `/var/solr/data` as it was for `server/solr`. So just remove this line?
```suggestion
```

**Review by janhoy — CHANGES_REQUESTED** [review]

**Comment by epugh** [comment]
@jebnix I'm sorry that some of these great doc improvements fell by the wayside...    Let's get them over the finish line!

**Comment by janhoy** [comment]
I'm removing the entire sub chapter `How can I mount a host directory as a data volume` (which this PR touches) in #2277 since I.M.O. that it is a good practice at all for an application like solr. It just confuses. So I'll close this one.

@jebnix Really appreciate all your doc patches and I also apologise they fell behind. The "stale" bot brought new attention to them. Hope you are not discouraged from contributing again.

**Pull request closed without merging**